### DISCUSSION TITLE
Improve API Token Permission

### DISCUSSION THREAD

**Discussion by andrasbacsai** [coollabsio/coolify#3682] [Improvement Requests]
Currently we have the following permissions for API tokens:
- Root
- Read-only
- View Sensitive

Someone suggested `redeploy only` token, that make sense.

What else do we need?

**Comment by wdhdev** [comment]
I would like to see fine-grain permissions such as:
- Per project permissions
	- Redeploying
	- Deploying
	- Creating new deployments
	- etc...

This would be very useful as I do not like giving full root access to an API token only used for redeploying a project...

**Comment by fschirinzi** [comment]
I think granular tokens (Team->Project->Env) are an important (security) feature.
I am not the OG, but here is a good description: https://github.com/coollabsio/coolify/discussions/9912
